### PR TITLE
fix crash when using clm lake and GFS PBL/sfclay

### PR DESCRIPTION
**Pull request by JiliDong-NOAA**
This PR is to fix the RRFS ensemble member crash when combining clm lake and GFS PBL/sfclay:

https://github.com/ufs-community/ccpp-physics/issues/141

A new flag is created to track new freezing lake ice and let sfc_diff to calculate stability variables over the new icy grids for GFS PBL in the second surface loop, which otherwise will remain as missing values.

### CODE REVIEW

**Review by JongilHan66 — APPROVED** [review]

File: physics/clm_lake.f90 [hunk 1]
```diff
@@ -268,7 +268,7 @@ SUBROUTINE clm_lake_run( &
          ! Atmospheric model state inputs:
          tg3, pgr, zlvl, gt0, prsi, phii, qvcurr, gu0, gv0, xlat_d, xlon_d,       &
          ch, cm, dlwsfci, dswsfci, oro_lakedepth, wind, rho0, tsfc,               &
-         flag_iter, ISLTYP, rainncprv, raincprv,                                  &
+         flag_iter, flag_lakefreeze, ISLTYP, rainncprv, raincprv,                     &
```
Comment (dustinswales) [inline physics/clm_lake.f90]: Super picky here, but please remove excess whitespace.

**Review by dustinswales — APPROVED** [review]
The changes look fine to me, just a small comment on the formatting

File: physics/clm_lake.f90 (hunk 1 above)
Comment (JiliDong-NOAA) [reply physics/clm_lake.f90]: done

**Review by Qingfu-Liu — APPROVED** [review]

File: physics/clm_lake.f90 [hunk 2]
```diff
@@ -754,6 +756,11 @@ SUBROUTINE clm_lake_run( &
                   weasd(i)      = weasdi(i)
                   snowd(i)      = snodi(c)                  ! surface_snow_thickness_water_equivalent_over_ice
 
+
+                  if (.not. icy(i)) then
+                     flag_lakefreeze(i)=.true.
```
Comment (tanyasmirnova) [inline physics/clm_lake.f90:761]: @JiliDong-NOAA Jili, should it be **if(icy(i)) then** when the lake grid point is frozen?

Comment (tanyasmirnova) [reply physics/clm_lake.f90:761]: @JiliDong-NOAA I see now, that this will be true only for new ice points. I agree with this change. Thank you.

**Review by tanyasmirnova — APPROVED** [review]

**Pull request merged**